src/v/datalake/local_parquet_file_writer.cc

```diff
@@ -46,8 +47,18 @@ local_parquet_file_writer::initialize(const iceberg::struct_type& schema) {
         co_return writer_error::file_io_error;
     }
 
+    ss::file_output_stream_options opts;
+    // Minimal buffer size to flush aggressively and avoid buffering in
+    // memory.
+    opts.buffer_size = 4_KiB;
+    // No write behind because
+    // - data is appended to the stream batch wise linearly.
+    // - avoids per writer waiter allocations in the ss stream implementation
+    opts.write_behind = 0;
```

Comment:
I am not sure i understand why the write behind is not beneficial here ? With write behind we can interleave computation with IO operations. Why here we disable it ? We could schedule a write, move on to translation instead of waiting for the write to complete.

Reply:
I think you are right, let me run a few more tests and reopen this.